crux-http-server/src/crux/http_server.clj

```diff
@@ -164,17 +164,22 @@
                                   :opt-un [::valid-time
                                            ::transact-time])))
 
+(defn- check-and-return-body [spec body-edn]
```

Comment:
I'd be tempted to call this using `doto` to avoid the `-and-` in the function name
e.g.
```clojure
(doto (body->edn request)
  (validate-or-throw ::query-map))
```

makes a guarantee to the reader that the validate function absolutely cannot alter the value it's being passed